pkg/apis/controlplane/v1beta1/conversion.go

```diff
@@ -44,3 +47,301 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 	}
 	return nil
 }
+
+func Convert_v1beta1_GroupMember_To_controlplane_GroupMember(in *GroupMember, out *controlplane.GroupMember, s conversion.Scope) error {
+	if in.Pod != nil {
+		out.Pod = &controlplane.PodReference{
+			Name:      in.Pod.Name,
+			Namespace: in.Pod.Namespace,
+		}
+	}
+	if in.ExternalEntity != nil {
+		out.ExternalEntity = &controlplane.ExternalEntityReference{
+			Name:      in.ExternalEntity.Name,
+			Namespace: in.ExternalEntity.Namespace,
+		}
+	}
+	var ports []controlplane.NamedPort
+	ips := make([]controlplane.IPAddress, len(in.Endpoints))
+	for i, ep := range in.Endpoints {
+		if i == 0 {
+			for _, p := range ep.Ports {
+				ports = append(ports, controlplane.NamedPort{
+					Protocol: controlplane.Protocol(p.Protocol), Port: p.Port, Name: p.Name,
+				})
+			}
+		}
+		ips[i] = controlplane.IPAddress(ep.IP)
+	}
+	out.Ports = ports
+	out.IPs = ips
+	return nil
+}
+
+func Convert_controlplane_GroupMember_To_v1beta1_GroupMember(in *controlplane.GroupMember, out *GroupMember, s conversion.Scope) error {
```

Comment:
The metadata (i.e. UID, name, namespace) are all missing, that should be why upgrade tests failed.
You could add `out.ObjectMeta = in.ObjectMeta`

Reply:
Thanks! It seems that it is indeed the issue... The ObjectMeta contains `ResourceVersion` field, so I was reluctant to add it for the conversion, thought that it would mess up things... Also it seems that it is only necessary for conversions of registered types `appliedtogroups`, `addressgroups` and their patch objects.

Reply:
Yes, only the registered types need it, I wrote the comment in wrong place.